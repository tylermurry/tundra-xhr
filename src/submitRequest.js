require('babel-polyfill');

export default async (capturedRequest, port, matched = true, RealXHR) => {
  // Sending in a fire-and-forget fashion.
  const request = new RealXHR(); // eslint-disable-line no-undef
  const requestUrl = `http://localhost:${port}/${matched ? 'requests' : 'requests/type/unmatched'}`;

  request.open('POST', requestUrl);
  request.setRequestHeader('Content-Type', 'application/json');
  request.send(JSON.stringify(capturedRequest));
};
