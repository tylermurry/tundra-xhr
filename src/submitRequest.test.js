import submitRequest from './submitRequest';

describe('submitRequest', () => {
  beforeEach(() => {
    const xmlHttpRequestMocks = {
      open: jest.fn(),
      send: jest.fn(),
      setRequestHeader: jest.fn(),
    };

    global.XMLHttpRequest = () => xmlHttpRequestMocks;
  });

  it('should send the matched request properly', () => {
    submitRequest({ request: 'data' }, 123, true, global.XMLHttpRequest);

    expect(global.XMLHttpRequest().open).toBeCalledWith('POST', 'http://localhost:123/requests');
    expect(global.XMLHttpRequest().setRequestHeader).toBeCalledWith('Content-Type', 'application/json');
    expect(global.XMLHttpRequest().send).toMatchSnapshot();
  });

  it('should send the unmatched request properly', () => {
    submitRequest({ request: 'data' }, 123, false, global.XMLHttpRequest);

    expect(global.XMLHttpRequest().open).toBeCalledWith('POST', 'http://localhost:123/requests/type/unmatched');
    expect(global.XMLHttpRequest().setRequestHeader).toBeCalledWith('Content-Type', 'application/json');
    expect(global.XMLHttpRequest().send).toMatchSnapshot();
  });
});
