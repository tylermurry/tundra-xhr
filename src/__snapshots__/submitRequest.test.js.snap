// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitRequest should send the matched request properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "{\\"request\\":\\"data\\"}",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`submitRequest should send the unmatched request properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "{\\"request\\":\\"data\\"}",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
